cmd/commands/common.go

```diff
@@ -247,11 +249,37 @@ func getIngressClassFromUserSelect(ctx context.Context, ingressClassNames []stri
 	return nil
 }
 
+func inferProviderFromRepo(repo string) (string, error) {
+	const suggest = "make sure the installation repository you have provided is at github or gitlab"
+
+	if strings.Contains(repo, "github.com") {
+		return "github", nil
+	}
+	if strings.Contains(repo, "gitlab.com") {
+		return "gitlab", nil
+	}
+
+	return "", fmt.Errorf("failed to infer git provider from repo url: %s. %s", repo, suggest)
+}
+
 func ensureGitToken(cmd *cobra.Command, cloneOpts *git.CloneOptions) error {
 	if cloneOpts.Auth.Password == "" && !store.Get().Silent {
-		return getGitTokenFromUserInput(cmd)
+		err := getGitTokenFromUserInput(cmd)
+		if err != nil {
+			return err
+		}
 	}
-	return nil
+
+	verified, err := cfgit.VerifyToken(cmd.Context(), cloneOpts.Provider, cloneOpts.Auth.Password)
+	if err != nil {
+		return fmt.Errorf("failed to verify git token: %w", err)
+	}
+
+	if verified {
+		return nil
+	}
+	
+	return fmt.Errorf("The provided git token is missing one or more of the required scopes")
```

Comment:
I think it could be huge if we actually print the scopes here. What I don't understand though is why it should even be here. Why don't the verifier return only error instead of bool and error. And the error can be formatted in each verifier.